test/org/zaproxy/zap/extension/api/OptionsParamApiUnitTest.java

```diff
@@ -0,0 +1,403 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2016 The ZAP development team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.api;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.isEmptyString;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.FileConfiguration;
+import org.junit.Test;
+import org.zaproxy.zap.utils.ZapXmlConfiguration;
+
+/**
+ * Unit test for {@link OptionsParamApi}.
+ */
+public class OptionsParamApiUnitTest {
+
+    private static final String API_ENABLED_KEY = "api.enabled";
+    private static final String API_SECURE_KEY = "api.secure";
+    private static final String API_KEY_KEY = "api.key";
+    private static final String API_DISABLEKEY_KEY = "api.disablekey";
+    private static final String API_INCERRORDETAILS_KEY = "api.incerrordetails";
+    private static final String API_AUTOFILLKEY_KEY = "api.autofillkey";
+    private static final String API_ENABLEJSONP_KEY = "api.enablejsonp";
+
+    @Test
+    public void shouldNotHaveConfigByDefault() {
+        // Given / When
+        OptionsParamApi param = new OptionsParamApi();
+        // Then
+        assertThat(param.getConfig(), is(equalTo(null)));
+    }
+
+    @Test
+    public void shouldHaveEnabledStateByDefault() {
+        // Given / When
+        OptionsParamApi param = new OptionsParamApi();
+        // Then
+        assertThat(param.isEnabled(), is(equalTo(true)));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldFailToSetEnabledWithoutConfig() {
+        // Given / When
+        OptionsParamApi param = new OptionsParamApi();
+        // When
+        param.setEnabled(true);
+        // Then = NullPointerException
+    }
+
+    @Test
+    public void shouldSetEnabledWithConfig() {
```

Comment:
Yeah, I initially had "EnabledState" but later changed to just "Enabled" (to match the other test names), this test is asserting that the "enabled state" is being changed/set (to false, since it's true by default).